python/ray/workflow/workflow_access.py

```diff
@@ -175,6 +177,13 @@ def run_or_resume(
         Returns:
             Workflow execution result that contains the state and output.
         """
+        # Re-configure log file to send logs to correct driver.
+        node = ray.worker._global_node
+        out_file, err_file = node.get_log_file_handles(
+            get_worker_log_file_name("WORKER", job_id)
+        )
+        configure_log_file(out_file, err_file)
+
```

Comment:
Let's not set it up here since this will change the actor's logs info.

Reply:
@iycheng I feel since we are setting this up in the actor's method (which always runs as `run_or_resume.remote()`), this log setting should only affect remote workers that does the actual tasks. And this should be fine as it does no impact on the actor worker and actor itself?

I could miss something, where else do you think might be a better place?

Reply:
This function is an actor method. Although it's a remote function, it's running in the actor class. I'm good about only updating `_resume_workflow_step_executor` for now.

Reply:
Think about this case, it's set to job1. And later there is other method called, and the log will be written to job1 as well which probably is not what we want.

Reply:
ah!!! I forgot the basic fact that WorkflowManagementActor is just a regular actor, so all the methods are scheduled on the same worker which is the actor's worker, so if I touch log config here, it will affect the actor.
Somehow I had this wrong feeling that whenever I see .remote(), it means it is placed on another worker.

Reply:
@iycheng do you think we might be able to do the following:
since `run_or_resume` is an immediately-returned and blocking (is it?) operations, can we do this in a `set up - tear down` fashion?  something like:


```
def run_or_resume():
        node = ray.worker._global_node
        original_out_file, original_err_file = node.get_log_file_handles(
            get_worker_log_file_name("WORKER")
        )
        new_out_file, new_err_file = node.get_log_file_handles(
                get_worker_log_file_name("WORKER", job_id)
            )
        configure_log_file(new_out_file, new_err_file)
        
        # body part here including logging

        configure_log_file(original_out_file, original_err_file)
```

Reply:
Right now, I'll say no because it's not an async actor. So, setup and tear down should work here. But given that there is not a lot of useful logs here, it should be ok not to cover this part.

The execution will be handled to this (https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/workflow/recovery.py?L226-228) and we've already set it up there.

Reply:
ok! I will remove it then.   And rebase with latest master and push it back once CI is back.

Reply:
@iycheng now I understand better on why you don't want to touch actor logs and I could see couple more scenarios where it can cause problem - one scenario is due to we have other "delayed" logs on actor like `report_success()` and `report_failure()`.

So if I do configure_logs in run_or_resume(), and suppose I have job_id_1 and job_id_2 both take 10 seconds to finish.  If job_id_2 (on driver 2) runs right after job_id_1 (on driver 1),  the job_id_1's successful log will be printed to job_id_2's concole. Because by the time job_id_1 finishes in 10 seconds and issues the "successful" log, job_id_2 has already triggered its configure_logs on the actor and changed the actor.

For the same reason, having setup - tear down in run_or_resume won't help this scenario either as it will tear down before the report_success is triggered, as tear down is triggered immediately after job is issued while report_success is triggered after the job fully finishes.

Ultimately if we really want the full log correctly send to the driver, we will have to add setup-tear-down to ALL places where the log could happen on the Actor, including small ones like report_success() and report_failure(), basically to pass job_id to all these places and use a close setup-tear-down to wrap every logging call.  It is doable but I agree with you - not too much useful information and maybe at this stage not worth the risk of touching the actor.